### ISSUE TITLE
Allow cronjobs with subminute schedule

### ISSUE THREAD

**Issue by ZronekM** [Der-Henning/tgtg#424]
Hello :)

First of all thank you for your great project. Just donated you 5 coffees :)

I configured my instance to run every minute between 6 and 22. Unfortunately though, some bags are so incredibly desired that they are sold out in less than 20 seconds and would therefore be . My preferred schedule would be to run the scans every 5 seconds for 1 minute after every quarter of an hour. (e.g. 6:00-6:01, 6:15-6:16, 6:30-6:31, ...). It's not very common here that bags are put online between the quarters.
I have seen that the following library supports subminute intervals: https://pypi.org/project/python-cron/
Is there a specific reason the other python library was taken?

**Comment by lars-devs** [comment]
Hi @ZronekM!

I got Your planned interval running as follows:

config.ini:
```
[MAIN]
SleepTime = 50
ScheduleCron = 45-46,50-51 15 * * *
```

Output:

```
2023-11-02 15:44:58 INFO     Active on schedule: At 45 through 46 and 50 through 51 minutes past the hour, at 15:00
2023-11-02 15:44:58 INFO     Scanner disabled by cron schedule.
2023-11-02 15:45:58 INFO     Scanner reenabled by cron schedule.
2023-11-02 15:47:38 INFO     Scanner disabled by cron schedule.
```

As long as Your bags aren't only available in quarters around 6 and in other hours more often, this cron should work for You:

`ScheduleCron = 0-1,15-16,30-31,45-46 6-22 * * *`

**Comment by ZronekM** [comment]
Hi @lars-devs ,

thanks for the cron. So the SleepTime and the ScheduleCron work together?

`[MAIN]
SleepTime = 5
ScheduleCron = 45-46,50-51 15 * * *`

This would check every 5 seconds?

**Comment by lars-devs** [comment]
> This would check every 5 seconds?

Yes, the sleep time triggers a scan regardless of the cron, but will then check if the current time fits in the cron schedule. So, in theory, there are 60 occasions which are valid if you scan with interval of 1 second and approx. 12 if Your interval is 5 seconds. But keep in mind, the scanner varies the scan interval slightly, see. 
`
sleep_time = self.config.sleep_time * (.9 + .2 * random())`

**Comment by ZronekM** [comment]
Thank you very much!
